testing/trino-tests/src/test/java/io/trino/tests/TestMockConnector.java

```diff
@@ -102,4 +116,45 @@ public void testDropMaterializedView()
     {
         assertUpdate("DROP MATERIALIZED VIEW mock.default.test_materialized_view");
     }
+
+    @Test
+    public void testDataGeneration()
+    {
+        assertQuery(
+                "SELECT * FROM mock.default.nation",
+                "SELECT * FROM nation");
+        assertQuery(
+                "SELECT nationkey FROM mock.default.nation",
+                "SELECT nationkey FROM nation");
+        assertQuery(
+                "SELECT nationkey, regionkey FROM mock.default.nation",
+                "SELECT nationkey, regionkey FROM nation");
+        assertQuery(
+                "SELECT regionkey FROM mock.default.nation",
+                "SELECT regionkey FROM nation");
+    }
+
+    @Test
+    public void testInsert()
+    {
+        assertUpdate("INSERT INTO mock.default.nation VALUES (101, 'POLAND', 0, 'No comment')", 1);
```

Comment:
nit: Assert that no actual INSERT took place

Reply:
+1
same for delete and update

Reply:
And comment why we do not expect INSERT/DELETE/UPDATE to actually happen